packages/zoom-in/src/components/zoom.js

```diff
@@ -0,0 +1,259 @@
+
+    const { overlay } = themeContext
+
+    scrollTop =
+      window.pageYOffset ||
+      document.documentElement.scrollTop ||
+      document.body.scrollTop ||
+      0
+
+    setAnimating(true)
+    containerRef.current.style.zIndex = overlay.zIndex
+    overlayRef.current.style.display = 'block'
+    window.requestAnimationFrame(() => {
+      overlayRef.current.style.opacity = overlay.opacity
+    })
+
+    zoomedRef.current.addEventListener('transitionend', handleOpenEnd)
+    setZoom(true)
+    animate({
+      originalRef,
+      zoomedRef,
+      captionRef,
+      themeContext,
+      captionHeight: getCaptionHeight(props.caption.length),
+    })
+  }
+
+  const handleCloseEnd = () => {
+    if (!zoomedRef.current) return
+
+    setZoom(false)
+    setAnimating(false)
+    containerRef.current.style.zIndex = '0'
+    overlayRef.current.style.display = 'none'
+    zoomedRef.current.removeEventListener('transitionend', handleCloseEnd)
+  }
+
+  const close = () => {
+    if (isAnimating || !isZoomed) return
+    if (!containerRef.current || !overlayRef.current || !zoomedRef.current)
+      return
+
+    setAnimating(true)
+    overlayRef.current.style.opacity = '0'
+    zoomedRef.current.addEventListener('transitionend', handleCloseEnd)
+    zoomedRef.current.style.transform = ''
+
+    captionRef.current.setAttribute(
+      'style',
+      `position: absolute;
+       top: 0;
+       left: 0;
+       opacity: 0;
+       visibility: hidden;`
+    )
+  }
+
+  const toggle = ({ target } = {}) => {
+    if (isZoomed) {
+      return close()
+    }
+    if (target) {
+      return open({ target })
+    }
+  }
+
+  const handleScroll = () => {
+    if (isAnimating) return
+
+    const currentScroll =
+      window.pageYOffset ||
+      document.documentElement.scrollTop ||
+      document.body.scrollTop ||
+      0
+
+    if (Math.abs(scrollTop - currentScroll) > zoomOptions.scrollOffset) {
+      setTimeout(close, 150)
+    }
+  }
+
+  const handleKeyUp = event => {
+    const key = event.key || event.keyCode
+
+    // Close if escape key is pressed
+    if (key === 'Escape' || key === 'Esc' || key === 27) {
+      close()
+    }
+  }
+
+  useEffect(() => {
+    if (!originalRef.current || !zoomedRef.current) return
+    originalRef.current.style.width = '100%'
+    zoomedRef.current.style.width = '100%'
+    overlayRef.current.style.display = 'none'
+  }, [])
+  useEventListener(document, 'keyup', handleKeyUp)
+  useEventListener(document, 'scroll', handleScroll)
+  useEventListener(window, 'resize', close)
+
+  if (Children.count(props.children) !== 1) {
+    console.error(`
+      Error: Adjacent JSX elements must be wrapped in an enclosing tag. 
+      Did you want a <div>...</div>?
+    `)
+    return null
+  }
+
+  const { overlay, caption } = themeContext
+
+  return (
+    <Container ref={containerRef}>
+      <Original isZoomed={isZoomed} onClick={toggle}>
+        {cloneElement(props.children, { ref: originalRef })}
```

Comment:
Question 🙋 :
Why do we pass `originalRef` to `props.children` rather than `Original` component?

Reply:
Please see [this commit](https://github.com/twreporter/orangutan-monorepo/pull/70/commits/71094dabd9885b3e04b732be3d96ccc214821943).

In this change, we are not passing `ref` or component to props.children.